### ISSUE TITLE
[BUG] <invMove conflicts with itemswapper>

### ISSUE THREAD

**Issue by surfishs** [tr7zw/ItemSwapper#160]
### Version information

0.4.0-1.20-farbic

### Other relevant installed mods

https://modrinth.com/mod/invmove

### Expected Behavior

Press the shortcut key to open the panel

### Actual Behavior

After pressing Shortcut, the panel quickly flashes and closes, unless you hold it down, it will display normally after a while.

### Reproduction Steps

install invmove，Press the itemswapper shortcut

### Render engine

- [ ] Vanilla
- [X] Sodium(optionally with Iris)
- [ ] Optifine
- [ ] VulkanMod
- [ ] Canvas
- [ ] Other

### Additional information

https://modrinth.com/mod/invmove

**Comment by tr7zw** [comment]
Unless you have toggle Mode activated, holding down the key is the intended behaviour. The flicker might be caused by the optifine fix, because optifine was broken with overlays that didn't have that behaviour.

**Comment by surfishs** [comment]
> 除非您激活了切换模式，否则按住该键是预期的行为。闪烁可能是由 optifine 修复引起的，因为 optifine 被不具有该行为的覆盖层破坏。

No, I'm running on fabric, no optifine.

**Comment by tr7zw** [comment]
Yes, but I had to change code so the mod will work with optifine, that has that flicker apparently as a side effect for some people. I don't know what exactly triggers the flicker to be visible.

**Comment by surfishs** [comment]
I've checked to determine that the error is caused by Invmove, which calls something about the backpack panel.
I have a bug demo video, but I can't send it to you here.

> 是的，但我必须更改代码，以便该 mod 可以与 optifine 一起使用，这对某些人来说显然是一个副作用。我不知道到底是什么触发了可见的闪烁。

**Comment by surfishs** [comment]
https://github.com/tr7zw/ItemSwapper/assets/96485650/6664be02-40b4-4547-ab08-85a06ecab6fa

**Comment by xR4YM0ND** [comment]
Same here, running Fabric 1.20.1 with Inv Move.

I want to open the Swapp Screen but i just flashes and closes when i press the hotkey.

**Comment by dani-tom** [comment]
![image](https://github.com/tr7zw/ItemSwapper/assets/135314441/10689d8a-1150-4bba-843a-9f736634826f)
The mod just doesn't work for me.

**Comment by grimid** [comment]
Having the same issue. Turning "Allow Walking With UI" off in itemswapper settings causes the item swapper ui to flicker very fast. Disabling invmove in mod settings doesn't fix it but removing invmove from your mods folder and relaunching fixes this issue.

**Comment by Zh40Le1ZOOB** [comment]
Same issue on Minecraft 1.21 with Fabric 0.16.0, but the issue disapperaed when in a multiplayer server.

**Comment by tr7zw** [comment]
Noteworthy that Mojang removed the flag to mark ui's that still allow you to move(basically what invMove and this mod needs). So now mods have conflicting solutions that potentially overwrite each other. Right now I don't have the time to look into this.

**Comment by Uzimattyan** [comment]
I have the same problem. But what is noteworthy is that in my case the problem is caused through [Amecs](https://modrinth.com/mod/amecs). That is, the bug only occurs when Amecs, InvMove, and ItemSwapper are present at the same time. And oddly enough, this problem does not occur with [too-many-shortcuts](https://modrinth.com/mod/too-many-shortcuts), a fork of Amecs.

**Comment by PieKing1215** [comment]
btw I'm looking into fixing this on InvMove's end (PieKing1215/InvMove#46)
Currently when InvMove sees a Screen that disables movement, it unpresses ALL keys (which is overkill)
So that causes issues when you're supposed to hold the key down for ItemSwapper
But I don't know why the issue only happens with Amecs/MouseWheelie for some people (including me)

And the reason InvMove sees the ItemSwapper overlay as blocking movement is because it uses a mixin to disable pausing in the IntegratedServer instead of overriding Screen::isPauseMenu
But I'm going to change InvMove to check Minecraft::isPaused instead so it'll work with the mixin way anyway

(tl;dr you don't need to change anything on your end if you don't want)

**Comment by tr7zw** [comment]
I can also switch it to use the Screen::isPauseMenu method with the next release (which is soon thanks to 1.21.2). Honestly probably just didn't see that method and instead made the mixin after Mojang removed the boolean in the screen class to still allow movement/not pausing.
